tests/components/yamaha/test_media_player.py

```diff
@@ -23,54 +26,142 @@ def __init__(self, ctrl_url, name, zones=None):
+        yield device
+
+
+async def test_setup_host(hass, device, main_zone):
+    """Test set up integration with host."""
+    assert await async_setup_component(hass, mp.DOMAIN, CONFIG)
+    await hass.async_block_till_done()
 
+    state = hass.states.get("media_player.yamaha_receiver_main_zone")
 
-class TestYamahaMediaPlayer(unittest.TestCase):
-    """Test the Yamaha media player."""
+    assert state is not None
+    assert state.state == "off"
 
-    def setUp(self):
-        """Set up things to be run when tests are started."""
-        self.hass = get_test_home_assistant()
-        self.main_zone = _create_zone_mock("Main zone", "http://main")
-        self.device = FakeYamahaDevice(
-            "http://receiver", "Receiver", zones=[self.main_zone]
+
+async def test_setup_no_host(hass, device, main_zone):
+    """Test set up integration without host."""
+    with patch("rxv.find", return_value=[device]):
+        assert await async_setup_component(
+            hass, mp.DOMAIN, {"media_player": {"platform": "yamaha"}}
         )
+        await hass.async_block_till_done()
+
+    state = hass.states.get("media_player.yamaha_receiver_main_zone")
+
+    assert state is not None
+    assert state.state == "off"
+
+
+async def test_setup_discovery(hass, device, main_zone):
+    """Test set up integration via discovery."""
+    discovery_info = {
+        "name": "Yamaha Receiver",
+        "model_name": "Yamaha",
+        "control_url": "http://receiver",
+        "description_url": "http://receiver/description",
+    }
+    await async_load_platform(
+        hass, mp.DOMAIN, "yamaha", discovery_info, {mp.DOMAIN: {}}
+    )
+    await hass.async_block_till_done()
+
+    state = hass.states.get("media_player.yamaha_receiver_main_zone")
+
+    assert state is not None
+    assert state.state == "off"
+
+
+async def test_setup_zone_ignore(hass, device, main_zone):
+    """Test set up integration without host."""
+    assert await async_setup_component(
+        hass,
+        mp.DOMAIN,
+        {
+            "media_player": {
+                "platform": "yamaha",
+                "host": "127.0.0.1",
+                "zone_ignore": "Main zone",
+            }
+        },
+    )
+    await hass.async_block_till_done()
+
+    state = hass.states.get("media_player.yamaha_receiver_main_zone")
+
+    assert state is None
+
+
+async def test_enable_output(hass, device, main_zone):
+    """Test enable output service."""
+    assert await async_setup_component(hass, mp.DOMAIN, CONFIG)
+    await hass.async_block_till_done()
+
+    port = "hdmi1"
+    enabled = True
+    data = {
+        "entity_id": "media_player.yamaha_receiver_main_zone",
+        "port": port,
+        "enabled": enabled,
+    }
+
+    await hass.services.async_call(DOMAIN, yamaha.SERVICE_ENABLE_OUTPUT, data, True)
+
+    assert main_zone.enable_output.call_count == 1
+    assert main_zone.enable_output.call_args == call(port, enabled)
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=5662368)